coalib/coala_main.py

```diff
@@ -69,6 +72,16 @@ def run_coala(log_printer=None,
         dtag = str(sections['default'].get('dtag', None))
         config_file = os.path.abspath(str(sections["default"].get("config")))
 
+        settings_hash = get_settings_hash(sections)
+        flush_cache = sections['default'].get('flush_cache', False)
+        if not flush_cache and settings_changed(log_printer, settings_hash):
```

Comment:
I'd love it if `settings_changed` could take the sections and not the hash thus making it easier to invoke and the main routine (which is really really too long) not longer than needed